ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/LinearTransactionCostCriterion.java

```diff
@@ -124,7 +127,7 @@ private Num getOrderCost(Order order, Num tradedAmount) {
      * @return the absolute total cost of all orders in the trade
      */
     private Num getTradeCost(TimeSeries series, Trade trade, Num initialAmount) {
-        Num totalTradeCost = series.numOf(0);
+        Num totalTradeCost = SeriesBuilder.getDefaultFunction().apply(0);
```

Comment:
So LTCC calculates only with the defaultFunction of `SeriesBuilder`? What if the `numFunction ` of `TimeSeries ` parameter is different? My first thought is to drop the Constructor initialization and only provide static `LTCC #calculate `functions with additional parameter `a`, `b` and `initialAmount` (here we have a time series with corresponding `numFunction`)

Reply:
I don't remember now why I made this change.  I think it had to do with a `ClassCastException` but when I undo that one line the tests run fine.  So I've changed it back.